carbonj.service/src/main/docker/files/check-invalid-namespaces.py

```diff
@@ -0,0 +1,184 @@
+        with open('/data/invalid_namespaces', 'a', buffering=1) as wf:
+            for file in files:
+                if stop:
+                    break
+                logger.info('Scanning IDs in file %s', file)
+                with open(file, 'r') as f:
+                    while True:
+                        line = f.readline()
+                        if not line:
+                            break
+                        key, value = get_key_value(line)
+                        name = bytes.fromhex(value[2:]).decode('utf-8')
+                        # Skip root check
+                        if name == 'root':
+                            continue
+                        if self.common_prefix != '.':
+                            if not name.startswith(self.common_prefix):
+                                continue
+                            prefix = '.'.join(name.split('.')[:self.prefix_parts])
+                            if prefix in checked_prefixes:
+                                # No need to check the same prefix again and again
+                                if invalid_id_count >= limit:
+                                    continue
+                            else:
+                                # Reset invalid ID count to check a new prefix
+                                invalid_id_count = 0
+                                checked_prefixes.append(prefix)
+                        first = name.split('.')[0]
+                        if first in self.skip_prefixes:
+                            continue
+
+                        logger.info('Checking namespace %s', name)
+                        valid_id = False
+                        for db_name in ['30m2y', '5m7d', '60s24h']:
+                            if check_id_in_db(db_name, key):
+                                valid_id = True
+                                break
+                        if not valid_id:
+                            wf.write(f"{name}\n")
+                            invalid_id_count += 1
+                            if invalid_id_count >= limit:
+                                if self.common_prefix == '.':
+                                    invalid_id_count = 0
+                                    self.skip_prefixes.append(first)
+                        else:
+                            self.skip_prefixes.append(first)
+
+    def cleanup_files(self):
+        files = glob.glob(f"/data/{self.db_name}_scan.*")
+        files.append('/data/invalid_namespaces')
+        for file in files:
+            try:
+                os.remove(file)
+            except Exception as e:
+                logger.error(f"Error: {e}")
+
+
+class Main:
+    def __init__(self):
+        return
+
+    def execute(self):
+        args = parse_args()
+        dumper = RocksDbKeyDumper('index-id', args)
+        dumper.dump()
+        dumper.check_invalid_ids()
+
+
+# Documentation https://salesforce.quip.com/gO3FAhYxHpQX
```

Comment:
Can you add instructions to the README instead of internal quip?

Reply:
I added the instruction in the quip doc